### PR TITLE
support concurrent-podgroup-syncs

### PR DESCRIPTION
**Pull request by WulixuanS**
A separate parameter to adjust the concurrency of pg sync
support #2996

### CODE REVIEW

**Comment by lowang-bh** [comment]
There is a param `WorkerThreads` to control it. Why do you add another one?

**Comment by WulixuanS** [comment]
> There is a param `WorkerThreads` to control it. Why do you add another one?
1. param `WorkerThreads` both controls job-controller and pg-controller. Now i used volcano to schedule pod, I only want to adjust pg-controller worker number to improve pg creation rate.
2. Each controller in K8S has independent syncs parameters to adjust, I think this is more flexible.

File: cmd/controller-manager/app/options/options.go [hunk 1]
```diff
@@ -60,6 +60,9 @@ type ServerOption struct {
 	EnableHealthz      bool
 	// To determine whether inherit owner's annotations for pods when create podgroup
 	InheritOwnerAnnotations bool
+	// ConcurrentPGSyncs is the number of pod groups that are allowed to sync concurrently.
+	// Larger number = faster pg creating, but more CPU (and network) load
+	ConcurrentPGSyncs uint32
```
Comment (william-wang) [inline cmd/controller-manager/app/options/options.go:65]: It's better unifiy the naming style for job and podgroup. Here is an example:

// WorkerThreadsForPG is the number of threads syncing podgroup operations
// The larger the number, the faster the podgroup processing, but requires more CPU load.
WorkerThreadsForPG

File: cmd/controller-manager/app/options/options.go [hunk 2]
```diff
@@ -90,6 +93,7 @@ func (s *ServerOption) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.HealthzBindAddress, "healthz-address", defaultHealthzAddress, "The address to listen on for the health check server.")
 	fs.BoolVar(&s.EnableHealthz, "enable-healthz", false, "Enable the health check; it is false by default")
 	fs.BoolVar(&s.InheritOwnerAnnotations, "inherit-owner-annotations", true, "Enable inherit owner annotations for pods when create podgroup; it is enabled by default")
+	fs.Uint32Var(&s.ConcurrentPGSyncs, "concurrent-podgroup-syncs", 1, "the number of pod groups that are allowed to sync concurrently. Larger number = faster pg creating, but more CPU (and network) load")
```
Comment (william-wang) [inline cmd/controller-manager/app/options/options.go:96]: The same as above

**Comment by william-wang** [comment]
> > There is a param `WorkerThreads` to control it. Why do you add another one?
> 
> 1. param `WorkerThreads` both controls job-controller and pg-controller. Now i used volcano to schedule pod, I only want to adjust pg-controller worker number to improve pg creation rate.
> 2. Each controller in K8S has independent syncs parameters to adjust, I think this is more flexible.
Would you share some testing data about improving pg create rate after adjust the worker threads?

**Review by william-wang — APPROVED** [review]
/lgtm

**Pull request merged**